File: examples/js/strops.js
// String length/substring and Math.trunc from real JavaScript — substring is
// the IR's substr, Math.trunc is an int cast, both verified against this run.
function pick(s, a, b) {
  return s.substring(a, b);
}

function main() {
  let s = "transpiler";
  console.log(s.length);
  console.log(pick(s, 0, 5));
  console.log(s.substring(5, s.length));
  console.log(Math.trunc(7.9));
  console.log("apple" < "banana");
}

main();

File: examples/strops.ml
func main(): void {
  let s: string = "transpile";
  print(len(s));
  print(substr(s, 0, 5));
  print(substr(s, 5, len(s)));

  print("apple" < "banana");
  print("b" < "apple");
  print("same" == "same");

  let w: string = "hello world";
  print(substr(w, 6, 11));
  print(len(substr(w, 0, 5)));
}
